pkg/destroy/aws/aws.go

```diff
@@ -660,70 +687,69 @@ func deleteEC2ElasticIP(ctx context.Context, client *ec2.EC2, id string, logger
 	return nil
 }
 
-func terminateEC2Instance(ctx context.Context, ec2Client *ec2.EC2, iamClient *iam.IAM, id string, logger logrus.FieldLogger) error {
+func terminateEC2Instance(ctx context.Context, ec2Client *ec2.EC2, iamClient *iam.IAM, id string, logger logrus.FieldLogger) (destroyedResources, error) {
```

Comment:
this function is supposed to destroy one instance, why does this function return destroyed resources, it's always this instance was destroyed or not?

Does it matter that the instance was not yet destroyed because the IAM instance profile couldn't be destroyed? The main resource is the instance that is still waiting to be destroyed.

personally I think this extra layer of tracking is not very informational for the user.